unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/CharacterController/DCLCharacterController.cs

```diff
@@ -346,11 +348,21 @@ internal void LateUpdate()
 
         if (isOnMovingPlatform)
         {
-            lastLocalGroundPosition = groundTransform.InverseTransformPoint(transform.position);
+            SaveLateUpdateGroundTransforms();
         }
 
         OnUpdateFinish?.Invoke(deltaTime);
     }
+    private void SaveLateUpdateGroundTransforms()
```

Comment:
I think we can add a test to ensure the character is rotated correctly? We already have tests in place to check moving platforms. Look for `CharacterSupportsRotatingPlatforms`. I think the test was made explicit but not sure why (maybe because of this bug).

Reply:
Added an assert to verify the rotation of the character after being rotated by the platform on the last commit, alas, this was working even without my changes, so an external system was causing the bug.